pkg/apis/serving/v1alpha2/framework_onnx.go

```diff
@@ -36,8 +41,15 @@ func (s *ONNXSpec) GetResourceRequirements() *v1.ResourceRequirements {
 }
 
 func (s *ONNXSpec) GetContainer(modelName string, parallelism int, config *InferenceServicesConfig) *v1.Container {
+	uri, _ := url.Parse(s.StorageURI)
+	var filename string
+	if path.Ext(uri.Path) != ONNXFileExt {
```

Comment:
I think we can validate if the file ext is `.onnx` otherwise reject and `DefaultONNXModelName` is no longer needed?

Reply:
I agree that would probably be more robust, but wouldn't that be a breaking change here https://github.com/kubeflow/kfserving/blob/67e3d1ac21da0ba5867ef2a88c4ff78ea35a6d94/docs/samples/pytorch/pytorch.yaml#L9? It's possible that people have not specified the actual file name in blob storage in their CRD but it still works at the present moment because the storage initializer will then list them out.

Reply:
My concern is that if user specified  some other ext e.g `gs://kfserving-samples/models/pytorch/cifar10/model.pkl` but we silently convert to model.onnx and in this case the load is going to fail for sure, so we may want to give that error up front.

Reply:
That sounds reasonable to me, better to give users helpful and clear error messages. The other case to consider is with the generic URI functionality, someone could also place their onnx model in a `zip` file:
```yaml
storageUri: s3://some_bucket/some/key/path/onnx_model.zip
```
or in a tarball. Do we want to reserve those MIME types to the models that require multiple artifacts (a la Tensorflow & Pytorch)?

Reply:
Currently S3 or GCS like object storage all does not support zip files in KFServing as it is not a common pattern to store models that way. We support zip for http in your last PR more for the reason that  http is not object store and it is also only for experiment and education purpose. I feel here we can validate if there is file ext then it needs to be `.onnx`, otherwise it is considered as model store and expect `model.onnx` under it, WDYT?

Reply:
Totally agree, just want to make sure I'm not introducing anything that leads to downstream frustrations for folks. I'll update it, thanks :smile: .

Reply:
updated in https://github.com/kubeflow/kfserving/pull/1078/commits/85d2a688113425c0b80a5b4c5dbe3b7763131c3e